ngraph/core/reference/include/ngraph/runtime/reference/roi_align.hpp

```diff
@@ -0,0 +1,239 @@
+                    std::vector<T> pooling_weights;
+
+                    pooling_points.reserve(4 * num_samples_in_bin * pooled_height * pooled_width);
+                    pooling_weights.reserve(4 * num_samples_in_bin * pooled_height * pooled_width);
+
+                    // Save the sample coords and weights as they will be identical across all
+                    // channels
+                    for (unsigned int y_bin_ind = 0; y_bin_ind < pooled_height; y_bin_ind++)
+                    {
+                        for (unsigned int x_bin_ind = 0; x_bin_ind < pooled_width; x_bin_ind++)
+                        {
+                            for (unsigned int y_sample_ind = 0; y_sample_ind < sampling_ratio;
+                                 y_sample_ind++)
+                            {
+                                T sample_y = y1 + static_cast<T>(y_bin_ind) * bin_height +
+                                             sample_distance_y * (static_cast<T>(y_sample_ind) +
+                                                                  static_cast<T>(0.5f));
+
+                                for (int64_t x_sample_ind = 0; x_sample_ind < sampling_ratio;
+                                     x_sample_ind++)
+                                {
+                                    T sample_x = x1 + static_cast<T>(x_bin_ind) * bin_width +
+                                                 sample_distance_x * (static_cast<T>(x_sample_ind) +
+                                                                      static_cast<T>(0.5f));
+
+                                    if (sample_x < -1.0 || sample_x > feature_map_width ||
+                                        sample_y < -1.0 || sample_y > feature_map_height)
+                                    {
+                                        // For this sample we save 4x point (0,0) with weight 0
+                                        pooling_points.insert(pooling_points.end(), 4, {0, 0});
+                                        pooling_weights.insert(pooling_weights.end(), 4, {0});
+                                        continue;
+                                    }
+
+                                    sample_x = std::max(sample_x, T{0});
+                                    sample_y = std::max(sample_y, T{0});
+
+                                    auto sample_y_low = static_cast<unsigned int>(sample_y);
+                                    auto sample_x_low = static_cast<unsigned int>(sample_x);
+                                    unsigned int sample_y_high;
+                                    unsigned int sample_x_high;
+
+                                    if (sample_y_low >= feature_map_height - 1)
+                                    {
+                                        sample_y_high = sample_y_low = feature_map_height - 1;
+                                        sample_y = (T)sample_y_low;
```

Comment:
Optional: earlier is used `static_case` style, I think that is better to keep uniformity

Reply:
I agree here, somehow this didn't bother me before